ilc/client.js

```diff
@@ -19,16 +23,30 @@ const appErrorHandlerFactory = (appName, slotName) => {
     return fragmentErrorHandlerFactory(registryConf, router.getCurrentRoute, appName, slotName);
 };
 
+const pluginManager = new PluginManager(require.context('./node_modules', true, /ilc-plugin-.*\/dist\/browser\.js$/));
 const i18n = registryConf.settings.i18n.enabled
     ? new I18n(registryConf.settings.i18n, {...singleSpa, triggerAppChange}, appErrorHandlerFactory)
     : null;
 const router = new Router(registryConf, state, i18n ? i18n.unlocalizeUrl : undefined, singleSpa);
+const guardManager = new GuardManager(router, pluginManager, internalErrorHandler);
+const urlProcessor = new UrlProcessor(registryConf.settings.trailingSlash);
+
+addNavigationHook((state) => {
+    if (state.nextUrl) {
```

Comment:
I have added it because it is in history update state methods can be undefined in some cases https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState (URL is optional)